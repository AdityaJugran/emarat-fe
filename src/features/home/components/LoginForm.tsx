import React, { useState, FormEvent } from 'react';
import { ConfirmationResult } from 'firebase/auth';
import { apiRequest } from 'app/apiRequest';
import { stripNonNumbers } from '../../../helpers/stringHelpers';
import { sendOtp, confirmOtp } from '../../../lib/firebaseAuth';
import Button from '../../common/Button';

export default function LoginForm() {
	const [mobile, setMobile] = useState<string>('');
	const [feedbackText, setfeedbackText] = useState<string>('');
	const [otpSentResult, setotpSentResult] = useState<ConfirmationResult>();

	const [otp, setOtp] = useState<string>('');

	const mobileHandler = (e: FormEvent<HTMLInputElement>) => {
		setMobile(stripNonNumbers(e.currentTarget.value));
	};

	const otpHandler = (e: FormEvent<HTMLInputElement>) => {
		setOtp(stripNonNumbers(e.currentTarget.value));
	};

	const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
		e.preventDefault();
		setfeedbackText('');

		if (mobile.length !== 10) {
			setfeedbackText('Mobile number must be 10 digits long.');
			return;
		}

		if (otpSentResult) {
			const jwt = await confirmOtp(otpSentResult, otp);

			if (jwt) {
				await apiRequest({ apiUrl: 'login', data: { token: jwt } });
			} else {
				setOtp('');
				setfeedbackText('Inavlid OTP entered.');
			}
		} else {
			const otpSentResultResult = await sendOtp(
				e.currentTarget,
				`+91${mobile}`
			);

			if (otpSentResultResult) {
				setotpSentResult(otpSentResultResult);
			} else {
				setfeedbackText('Could not send OTP.');
			}
		}
	};

	return (
		<div className="shadow-around px-7 py-10 rounded-3xl w-full">
			<div>
				<h3 className="text-4xl font-regular">Login</h3>
				<p className="mt-2 text-gray-400">
					Login to your Emarat account
				</p>
				<form onSubmit={handleSubmit}>
					{otpSentResult ? (
						<input
							type="text"
							className="block mt-6 border py-3 px-4 border-gray-200 rounded-lg w-full outline-none"
							placeholder="OTP"
							onInput={otpHandler}
							value={otp}
							maxLength={6}
						/>
					) : (
						<input
							type="text"
							className="block mt-6 border py-3 px-4 border-gray-200 rounded-lg w-full outline-none"
							placeholder="Mobile Number"
							onInput={mobileHandler}
							value={mobile}
							maxLength={10}
						/>
					)}
					<div className="mt-1 pl-1 text-xs text-red-500">
						{feedbackText}
					</div>
					<Button></Button>
				</form>
			</div>
		</div>
	);
}
