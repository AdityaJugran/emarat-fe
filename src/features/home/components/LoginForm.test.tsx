import renderer from 'react-test-renderer';
import LoginForm from 'features/home/components/LoginForm';

it('Login Form', () => {
	const tree = renderer.create(<LoginForm />).toJSON();
	expect(tree).toMatchSnapshot();
});
