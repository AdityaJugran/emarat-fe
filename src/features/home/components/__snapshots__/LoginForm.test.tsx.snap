// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form 1`] = `
<div
  className="shadow-around px-7 py-10 rounded-3xl w-full"
>
  <div>
    <h3
      className="text-4xl font-regular"
    >
      Login
    </h3>
    <p
      className="mt-2 text-gray-400"
    >
      Login to your Emarat account
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        className="block mt-6 border py-3 px-4 border-gray-200 rounded-lg w-full outline-none"
        maxLength={10}
        onInput={[Function]}
        placeholder="Mobile Number"
        type="text"
        value=""
      />
      <div
        className="mt-1 mb-6 pl-1 text-xs text-red-500"
      >
        
      </div>
      <button
        className="sc-bdfBQB dzClhw text-white bg-emarat-accent-default hover:bg-emarat-accent-hover active:bg-emarat-accent-active rounded-full"
        disabled={false}
        type="submit"
      >
        Send OTP
      </button>
    </form>
  </div>
</div>
`;
